src/main/java/io/tiledb/spark/TileDBDataReaderPartitionScan.java

```diff
@@ -318,15 +326,49 @@ private boolean initQuery() throws TileDBError {
     query = new Query(array, QueryType.TILEDB_READ);
 
     // Pushdown any ranges
-    if (pushedRanges.size() > 0) {
-      for (List<Range> ranges : pushedRanges) {
+    QueryCondition finalCondition = null;
+    if (allRanges.size() > 0) {
+      int j = 0;
+      // the first element of the allranges list is a list of the dimension ranges. The remaining
```

Comment:
This logic could be simplified with (IIUC) no overhead by taking two sublists -- the first element and the trailing elements -- and handle the dimension and QC cases in separate `for` loops.

Reply:
Done

Reply:
Much easier to read now, IMO -- thank you!